src/sdk/pynni/nni/compression/torch/compressor.py

```diff
@@ -272,10 +286,19 @@ class Pruner(Compressor):
 
     """
 
-    def __init__(self, model, config_list):
-        super().__init__(model, config_list)
+    def __init__(self, model, config_list, optimizer):
+        super().__init__(model, config_list, optimizer)
+        self.patch_optimizer(self.update_mask)
 
-    def calc_mask(self, layer, config, **kwargs):
+    def compress(self):
+        self.update_mask()
+        return self.bound_model
+
+    def update_mask(self):
+        for wrapper in self.get_modules_wrapper():
+            self.calc_mask(wrapper)
```

Comment:
better to update masks after `calc_mask()`

Reply:
updated